### PR TITLE
Add SSHKeys module

### PR DESCRIPTION
**Pull request by mazz-seven**

### CODE REVIEW

File: lib/digital_ocean/ssh_keys.ex [hunk 1]
```diff
@@ -0,0 +1,96 @@
+defmodule DigitalOcean.SSHKeys do
```
Comment (anthonator) [inline lib/digital_ocean/ssh_keys.ex:1]: Resource modules should be singular. Could you please rename to `SSHKey`? Also, could you please rename the file to `ssh_key.ex`?

File: lib/digital_ocean/ssh_keys.ex [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  alias DigitalOcean.{ Operation }
+
+  @doc """
+  Create a new SSH key.
+
+  ## Example creating a new SSH key
```
Comment (anthonator) [inline lib/digital_ocean/ssh_keys.ex:7]:
```suggestion
  ## Examples
```

File: lib/digital_ocean/ssh_keys.ex [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+      iex> DigitalOcean.SSHKeys.create(
+      ...>   name: "my ssh public key",
+      ...>   public_key: "ssh-rsa AEXAMPLEaC1yc2EAAAADAQABAAAAQQDDHr/jh2Jy4yALcK4JyWbVkPRaWmhck3IgCoeOO3z1e2dBowLh64QAM+Qb72pxekALga2oi4GvT+TlWNhzPH4V example",
+      ...> ) |> DigitalOcean.request()
+      { :ok, %DigitalOcean.Response{} }
+  """
+  @type create_opts :: [name: String.t(), public_key: String.t()]
```
Comment (anthonator) [inline lib/digital_ocean/ssh_keys.ex:15]: We don't want to be responsible for the arguments sent to the API. If they change then we'll cause dialyzer errors for developers using this library that use arguments not present in the type. So please remove.

File: lib/digital_ocean/ssh_keys.ex [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  @spec create(create_opts()) :: Operation.t()
```
Comment (anthonator) [inline lib/digital_ocean/ssh_keys.ex:16]:
```suggestion
  @spec create(Keyword.t()) :: Operation.t()
```

File: lib/digital_ocean/ssh_keys.ex [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    |> Map.put(:method, :get)
+    |> Map.put(:path, "/account/keys/#{key_id}")
+  end
+
+  @doc """
+  Retrieve a list of all SSH keys.
+  ## Examples
+      iex> DigitalOcean.SSHKeys.list() |> DigitalOcean.request()
+      { :ok, %DigitalOcean.Response{} }
+  """
```
Comment (anthonator) [inline lib/digital_ocean/ssh_keys.ex:50]:
```suggestion
  @doc """
  Retrieve a list of all SSH keys.
  
  ## Examples
  
      iex> DigitalOcean.SSHKeys.list() |> DigitalOcean.request()
      { :ok, %DigitalOcean.Response{} }
  """
```

File: lib/digital_ocean/ssh_keys.ex [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+      iex> DigitalOcean.SSHKeys.update(512189, name: "New Name For SSH Public Key") |> DigitalOcean.request()
+      { :ok, %DigitalOcean.Response{} }
+
+  ## Examples
+
+      iex> DigitalOcean.SSHKeys.update("3b:16:bf:e4:8b:00:8b:b8:59:8c:a9:d3:f0:19:45:fa", name: "New Name For SSH Public Key") |> DigitalOcean.request()
+      { :ok, %DigitalOcean.Response{} }
+  """
+  @typep update_opts :: [name: String.t()]
```
Comment (anthonator) [inline lib/digital_ocean/ssh_keys.ex:73]: We don't want to be responsible for the arguments sent to the API. If they change then we'll cause dialyzer errors for developers using this library that use arguments not present in the type. So please remove.

File: lib/digital_ocean/ssh_keys.ex [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  @spec update(DigitalOcean.id_t(), update_opts()) :: Operation.t()
```
Comment (anthonator) [inline lib/digital_ocean/ssh_keys.ex:74]:
```suggestion
  @spec update(DigitalOcean.id_t(), Keyword.t()) :: Operation.t()
```

File: lib/digital_ocean/ssh_keys.ex [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  end
+
+  @doc """
+  Delete a droplet.
+
+  ## Examples
+
+      iex> DigitalOcean.Droplet.delete(3164494) |> DigitalOcean.request()
+      { :ok, %DigitalOcean.Response{} }
+  """
```
Comment (anthonator) [inline lib/digital_ocean/ssh_keys.ex:89]: Please update for this module.

File: lib/digital_ocean/ssh_keys.ex [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+  def list(opts \\ []) do
+    %Operation{}
+    |> Map.put(:method, :get)
+    |> Map.put(:params, opts)
+    |> Map.put(:path, "/account/keys")
+  end
+
+  @doc """
+  Update the name of an SSH key.
+  Either the ID or the fingerprint of an existing SSH key
```
Comment (anthonator) [inline lib/digital_ocean/ssh_keys.ex:61]:
```suggestion
```

**Comment by mazz-seven** [comment]
Took me a while to get back to it. :)

**Comment by anthonator** [comment]
Nice! Thank you! I was actually about to close this PR since it was stale. I'll do a review later today.

**Pull request merged**